### ISSUE TITLE
[Unified Map] Blinking/hidden Text labels

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#15400]
### Describe your problem!

When sending Unified Map to background for a while (e. g. by opening a listing page via cache icon on map or opening another app) and then returning to the map (resp. switching back to c:geo), text labels start to blink sometimes:

https://github.com/cgeo/cgeo/assets/9889015/cb1f3885-3f4e-4d54-bb90-84925e68a76a

This bug is so much annoying that I had to deactivate UM again, as it simply became unusable after a while (plus, as second reason, re-routing takes ages after a while - see below).  Nevertheless, most regretfully, I was not able to nail this behavior down to a reproducible scenario.

Usually I would not even think about opening an issue without a clear scenario, but since this bug is so annoying, I need to make an exception this time. Perhaps someone else has already stumbled upon this, too, and even got a deterministic way to make it happen.

For now I can just say, probability for flicker *seems* to increase, the longer UM resp. c:geo is running in background ... but, as said before, no real rule, just an impression.  :(

BTW: Same (still) applies to the time needed to recalculate a route after changing the routing style (e. g. from "Car" to "Bike"), which sometimes takes *so* long that I get in doubt if c:geo even realized the change - then I usually even kill c:geo in order to get the new route (seems as if the longer you use the app, the longer recalculation takes).  But this is a non-deterministic thing, too, so I just wanted to mention it over here, too - if s/o has observed the same (and has a way to reproduce) I'll be happy to make a dedicated issue for it).

### How to reproduce?

1. open Unified map
2. preferably, start navigation via map icon
3. drive a while
4. open any cache via map icon (resp. change to a different app)
5. stay a while in the listing page (resp. the other app)
6. go back to c:geo's Unified Map 

### Actual result after these steps?

Titles flicker - see video

### Expected result after these steps?

Things should be displayed normally.

### Reproducible

Unclear

### c:geo Version

2024.03.01-NB-ea8592e

**Comment by MagpieFourtyTwo** [comment]
Addition: Once the labels start to blink, the only way to get rid of this is, to completely close, remove from memory and restart c:geo.

**Comment by moving-bits** [comment]
This might be the same (or may have the same cause) as described in https://github.com/cgeo/cgeo/issues/15100#issuecomment-1962972600

**Comment by MagpieFourtyTwo** [comment]
And here's another "graphical glitch":

https://github.com/cgeo/cgeo/assets/9889015/7f04544d-0502-46ea-bb1d-30a8376c674a

Don't wonder, it's just a cropped version of the screen recording, as there are solved mysteries in this area. But I can of course send the complete video if needed.

**Comment by MagpieFourtyTwo** [comment]
> This might be the same (or may have the same cause) as described in [#15100 (comment)](https://github.com/cgeo/cgeo/issues/15100#issuecomment-1962972600)

Yep - seems to be the same one.  Wonder why I did not find it when searching for a matching issue ...
So, feel free to close this one, if it's not of any additional help.

**Comment by moving-bits** [comment]
Interesting question is, though, if those glitches only appear after you have opened a "second map", meaning
- started from map
- calling some sub activity (eg: cache details)
- from which you open another map (eg: by routing to that cache) without having actively closed the first one

or if one or both glitches happen to you even if you don't open such a sub-map.

**Comment by MagpieFourtyTwo** [comment]
Already thought about this after reading through the issue you mentioned above.
Can't swear, but would say that this even happened *without* opening a second map.
I remember two scenarios pretty well, I would say:

1. it even happened after switching to a different app (checked traffic) and, after a while, returning to c:geo via ``Recent apps``
2. in at least one case I can remember I just opened a cache listing via map icon, ready through the listing, and then directly went back to the map cause I needed to see the already active route to the cache we were heading to before

But, to be honest, can't fully remember how things went, and I can't reproduce it right now, thus I will have an eye on it and give some more feedback once I know it for sure.
